rushil-118/ChessGame
# Implemented Game over Logic

# Implemented Game over Logic 


## Description
- **Purpose:** This PR implements the game over logic for the chess game. When the king piece is killed, all pieces are cleared from the board, and the game is marked as over.

- **Changes Made:**
  - Updated the `kill` method to check if the killed piece is a king.
  - Implemented the `endGame` method to remove the king from the board and display a game over message.
  
  ## Motivation and Context
- The game over logic is essential for indicating the end of the game when a king is captured. This change ensures that the game behaves correctly and provides feedback to the players.

## How to Test
1. Set up the chess game and move pieces until the king is captured.
2. Verify that the game stops, the king piece is removed from the board, and the correct game over message is displayed.


## Screenshots
![image](https://github.com/user-attachments/assets/737f377f-2f99-4627-b224-e54f9887254e)

# Optional Update
I expect the winner or the currentplayer who won ,to be displayed in the noticeBoard below the chess board
